### PR TITLE
Better error message when client specifies improper connection information

### PR DESCRIPTION
**Pull request by broomweed**
And also a bunch of library files got added by IntelliJ, somehow.

### CODE REVIEW

File: .idea/sbt.xml [hunk 1]
```diff
@@ -10,9 +10,8 @@
             <option value="$PROJECT_DIR$/project" />
           </set>
         </option>
-        <option name="sbtVersion" value="1.0.3" />
+        <option name="sbtVersion" value="1.0.4" />
         <option name="useAutoImport" value="true" />
-        <option name="useOurOwnAutoImport" value="true" />
```
Comment (mcoblenz) [inline .idea/sbt.xml]: Why the change here?

File: .idea/.name [hunk 2]
```diff
@@ -1 +1 @@
-Obsidian
```
Comment (mcoblenz) [inline .idea/.name]: Why is this change here?

File: .idea/libraries/sbt__com_google_protobuf_protobuf_java_3_5_1_jar.xml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<component name="libraryTable">
```
Comment (mcoblenz) [inline .idea/libraries/sbt__com_google_protobuf_protobuf_java_3_5_1_jar.xml]: Actually I looked at this and I think we should ignore these files (gitignore) instead: https://intellij-support.jetbrains.com/hc/en-us/articles/206544839-How-to-manage-projects-under-Version-Control-Systems

**Comment by mcoblenz** [comment]
Also, this change should be split into two: one to fix the error message and one to fix the library files.

**Comment by broomweed** [comment]
OK, I'm going to close this one and create two new branches instead.
